### PR TITLE
SqlServer: Drop default constraints during add/alter column operations

### LINKED ISSUE #1345: Reverting migration that adds column with default value throws
When attempting to revert a migration that adds an int column with a default value, migrations throws a SqlException.

Migration code:
```cs
    public partial class AddValue : Migration
    {
        public override void Up(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.AddColumn("MyData", "Value", c => c.Int(nullable: false, defaultValue: 2));
        }
        
        public override void Down(MigrationBuilder migrationBuilder)
        {
            migrationBuilder.DropColumn("MyData", "Value");
        }
    }
```

Error:
```
> k ef migration apply 0
Using context 'MyContext'.
Reverting migration '201501021757503_AddValue'.
System.Data.SqlClient.SqlException (0x80131904): The object 'DF__MyData__Value__25869641' is dependent on column 'Value'.
ALTER TABLE DROP COLUMN Value failed because one or more objects access this column.
```

Script:
```
> k ef migration script AddValue InitialCreate
Using context 'MyContext'.
ALTER TABLE [MyData] DROP COLUMN [Value]
DELETE FROM [__MigrationHistory] WHERE [MigrationId] = '201501021757503_AddValue' AND [ContextKey] = 'MigrationDownRepro.MyContext'
```

### PR DESCRIPTION
**Pull request by bricelam**
Fixes #1345

### CODE REVIEW

File: src/EntityFramework.SqlServer/Migrations/SqlServerMigrationSqlGenerator.cs [hunk 1]
```diff
@@ -36,12 +37,38 @@ public class SqlServerMigrationSqlGenerator : MigrationSqlGenerator
...
                 .Append(" ALTER COLUMN ");
-            ColumnDefinition(operation, model, builder);
+            ColumnDefinition(
+                    operation.Schema,
+                    operation.Table,
+                    operation.Name,
+                    operation.ClrType,
+                    operation.ColumnType,
+                    operation.IsNullable,
+                    /*defaultValue:*/ null,
```
Comment (anpete) [inline src/EntityFramework.SqlServer/Migrations/SqlServerMigrationSqlGenerator.cs]: Why comment these?

**Comment by anpete** [comment]
:shipit:

File: test/EntityFramework.SqlServer.FunctionalTests/MigrationsSqlServerTest.cs [hunk 2]
```diff
@@ -11,5 +15,95 @@ public MigrationsSqlServerTest(MigrationsSqlServerFixture fixture)
...
+            Assert.Equal(
+                @"
+CreatedTable
+    Id int NOT NULL
+    ColumnWithDefaultToAlter int NULL
+",
+                sql);
+        }
+
+        private async Task<string> GetDatabseSchemaAsync(DbConnection connection)
```
Comment (natemcmaster) [inline test/EntityFramework.SqlServer.FunctionalTests/MigrationsSqlServerTest.cs]: Nit: typo - missing an a

**Comment by natemcmaster** [comment]
LGTM :+1:

**Pull request merged**